### PR TITLE
feat: [CI-4949]: Arm pipeline commit

### PR DESCRIPTION
**Pull request by jamie-harness**
## Describe your changes

change the linux image build. The binary is now called only ci-addon-linux. Made changes on CI manager accordingly also

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36426)

### CODE REVIEW

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 1]
```diff
@@ -9,21 +9,21 @@
 # > docker build -t harness/ci-addon:tag1.0 -f product/ci/addon/Dockerfile $(bazel info bazel-bin)/product/ci/addon/
 
 
-FROM redhat/ubi8-minimal
+FROM --platform=linux/arm64 redhat/ubi8-minimal
```
Comment (shubham149) [inline product/ci/addon/docker/Dockerfile.linux.arm64:12]: Platform can be specified at docker build time.

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 2]
```diff
@@ -9,21 +9,21 @@
...
 RUN microdnf update \
     && microdnf install shadow-utils \
     curl \
     && rm -rf /var/cache/yum \
     && microdnf clean all
 
 # Copy ci-addon binary
-COPY ci-addon_/ci-addon /usr/local/bin/ci-addon-linux-arm64
+COPY ci-addon_/ci-addon /usr/local/bin/ci-addon-linux-amd64
```
Comment (shubham149) [inline product/ci/addon/docker/Dockerfile.linux.arm64:20]: this needs to be arm64. It is confusing right now.

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 3]
```diff
@@ -9,21 +9,21 @@
...
 COPY java-agent.jar /usr/local/bin/
 COPY split_tests /usr/local/bin/
-RUN chmod +x /usr/local/bin/ci-addon-linux-arm64 \
+RUN chmod +x /usr/local/bin/ci-addon-linux-amd64 \
```
Comment (shubham149) [inline product/ci/addon/docker/Dockerfile.linux.arm64:23]: Ditto

File: product/ci/addon/docker/Dockerfile.linux.arm64 [hunk 4]
```diff
@@ -9,21 +9,21 @@
...
     && chmod +x /usr/local/bin/split_tests
 
 RUN groupadd harness && adduser -g harness -s /bin/sh -u 1000 harness
 USER harness:harness
 
-CMD ["/usr/local/bin/ci-addon-linux-arm64"]
+CMD ["/usr/local/bin/ci-addon-linux-amd64"]
```
Comment (shubham149) [inline product/ci/addon/docker/Dockerfile.linux.arm64:29]: Ditto

File: product/ci/engine/docker/Dockerfile.linux.arm64 [hunk 5]
```diff
@@ -4,7 +4,7 @@
 # > bazel build --workspace_status_command=/bin/true --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //product/ci/engine:engine
 # > docker build -t harness/ci-lite-engine:<tag> -f product/ci/engine/Dockerfile $(bazel info bazel-bin)/product/ci/engine/
 
-FROM redhat/ubi8-minimal
+FROM --platform=linux/arm64 redhat/ubi8-minimal
```
Comment (shubham149) [inline product/ci/engine/docker/Dockerfile.linux.arm64:7]: Ditto

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36426)

**Review by shubham149 — APPROVED** [review]
> Reviewed 2 of 4 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36426)* status: :shipit: complete! ✓

**Pull request merged**